apis/projectcontour/v1/httpproxy.go

```diff
@@ -171,6 +171,26 @@ type AuthorizationServer struct {
 	//
 	// +optional
 	FailOpen bool `json:"failOpen,omitempty"`
+
+	// BufferSettings specifies configuration for buffering request data sent to AuthorizationServer
+	// +optional
+	BufferSettings *AuthorizationServerBufferSettings `json:"bufferSettings,omitempty"`
+}
+
+// AuthorizationServerBufferSettings enables ExtAuthz filter to buffer client request data and send it as part of authorization request
+type AuthorizationServerBufferSettings struct {
+	// MaxRequestBytes sets the maximum size of message body ExtAuthz filter will hold in-memory.
+	// +optional
+	// ++kubebuilder:validation:Minimum=1
```

Comment:
```suggestion
	// +kubebuilder:validation:Minimum=1
```

will need a regenerate of the crds as well